core/src/main/java/org/infinispan/remoting/MIMECacheEntry.java

```diff
@@ -47,4 +55,35 @@ public boolean equals(Object o) {
    public int hashCode() {
       return 31 * (contentType != null ? contentType.hashCode() : 0) + (data != null ? Arrays.hashCode(data) : 0);
    }
+
+   public static class Externalizer extends AbstractExternalizer<MIMECacheEntry> {
+
+      @Override
+      public Set<Class<? extends MIMECacheEntry>> getTypeClasses() {
+         return Util.<Class<? extends MIMECacheEntry>>asSet(MIMECacheEntry.class);
+      }
+
+      @Override
+      public void writeObject(ObjectOutput out, MIMECacheEntry obj) throws IOException {
+         out.writeUTF(obj.contentType);
+         out.writeInt(obj.data.length);
+         out.write(obj.data);
```

Comment:
I can certainly use `marshallByteArray()` but I'd argue that the `marshallInt` call inside is doing extra work that's not required specifically for byte[]. E.g. there's no need to check for any negative numbers.

On top of that, since you're talking about duplication, what does `MarshallUtil.marshallInt` do differently to `UnsignedNumeric.writeUnsignedInt`? Is the difference dealing with negative numbers or is there more to it?